apps/site/assets/ts/schedule/components/schedule-finder/UpcomingDepartures.tsx

```diff
@@ -7,23 +7,26 @@ import {
 import { modeIcon } from "../../../helpers/icon";
 import { modeBgClass } from "../../../stop/components/RoutePillList";
 import { Route } from "../../../__v3api";
-import { StopPrediction } from "../__schedule";
+import { EnhancedJourney } from "../__trips";
 import { breakTextAtSlash } from "../../../helpers/text";
+import { Accordion } from "./TableRow";
+import { UserInput } from "../ScheduleFinder";
 
 interface State {
-  data: StopPrediction[] | null;
+  data: EnhancedJourney[] | null;
   isLoading: boolean;
   error: boolean;
 }
 
 interface Props {
   state: State;
+  input: UserInput;
 }
 
-const hasPredictions = (stopPredictions: StopPrediction[]): boolean =>
-  stopPredictions.filter(
-    stopPrediction => stopPrediction.prediction.prediction !== null
-  ).length > 0;
+// Predictions are nil unless they have a time. This helps
+// prevent far-future trips from appearing in Upcoming Departures.
+const hasPredictions = (journeys: EnhancedJourney[]): boolean =>
```

Comment:
I think this function could be more clearly named to indicate that it both accepts and returns an array of journeys. `journeysWithPredictions`? (maybe that sounds too much like we're synthesizing Journey objects and Prediction objects...)

Reply:
I think you may be misreading this one, it's returning a boolean, not an array of journeys.

Reply:
Hm... not sure anymore what I was thinking, as it's now 14 days later 😅